Modules/ExtendedPlayerControl.cs

```diff
@@ -249,17 +251,72 @@ public static void RpcBootFromVentDesync(this PlayerPhysics physics, int ventId,
         writer.WritePacked(ventId);
         AmongUsClient.Instance.FinishRpcImmediately(writer);
     }
-    public static void RpcSetRoleDesync(this PlayerControl player, RoleTypes role, bool canOverride, int clientId)
+
+    public static void RpcRevive(this PlayerControl player, RoleTypes roleTypes, bool IsDesyncImpostor = false, List<PlayerControl> FellowImps = null)
+    {
+        if (player.Data.IsDead == false || roleTypes is RoleTypes.GuardianAngel or RoleTypes.CrewmateGhost or RoleTypes.ImpostorGhost)
+        {
+            Logger.Warn($"Invalid Revive for {player.GetRealName()} of roletype: {roleTypes} / Player was already alive? {!player.Data.IsDead}", "RpcRevive");
+            return;
+        }
+
+        player.RpcChangeRoleBasis(roleTypes, IsDesyncImpostor, FellowImps);
+        Main.PlayerStates[player.PlayerId].IsDead = false;
+        player.SyncGeneralOptions();
```

Comment:
Btw after revive need do `SetKillCooldown();`

Reply:
o right

Reply:
I'll also have to do this for desync crew in intropatch